src/superphot_plus/classify_ztf.py

```diff
@@ -23,180 +30,321 @@
+                plot_metrics=True,
+            )
 
-    true_classes_mlp = np.array([])
-    predicted_classes_mlp = np.array([])
-    prob_above_07_mlp = np.array([], dtype=bool)
+            # Test model on remaining data
+            test_classes, test_names, pred_classes, pred_probs = model.evaluate(
+                test_data=TestData(test_features, test_classes, test_names, test_group_idxs),
+                probs_csv_path=csv_path,
+            )
 
-    def run_single_fold(x):
-        train_index, test_index = x
-        train_labels = labels[train_index]
-        test_labels = labels[test_index]
-        test_names = np.array(names[test_index])
+            return pred_classes, pred_probs > 0.7, test_classes, test_names, best_valid_loss
+
+        r = Parallel(n_jobs=-1)(delayed(run_single_fold)(i, fold) for i, fold in enumerate(kfold))
+        (
+            predicted_classes_mlp,
+            prob_above_07_mlp,
+            true_classes_mlp,
+            ztf_test_names,
+            valid_loss_mlp,
+        ) = zip(*r)
+
+        predicted_classes_mlp = np.hstack(tuple(predicted_classes_mlp))
+        prob_above_07_mlp = np.hstack(tuple(prob_above_07_mlp))
+        true_classes_mlp = np.hstack(tuple(true_classes_mlp))
+        ztf_test_names = np.hstack(tuple(ztf_test_names))
+        valid_loss_avg = np.mean(valid_loss_mlp)
+
+        true_classes_mlp = SnClass.get_labels_from_classes(true_classes_mlp)
+        predicted_classes_mlp = SnClass.get_labels_from_classes(predicted_classes_mlp)
+
+        test_acc = calc_accuracy(predicted_classes_mlp, true_classes_mlp)
+        test_f1_score = f1_score(predicted_classes_mlp, true_classes_mlp, class_average=True)
+
+        self.plot_matrices(
+            num_epochs=num_epochs,
+            true_classes=true_classes_mlp,
+            predicted_classes=predicted_classes_mlp,
+            prob_above_07=prob_above_07_mlp,
+        )
 
-        train_index, val_index = train_test_split(train_index, stratify=train_labels, test_size=0.1)
+        if extract_wc:
+            extract_wrong_classifications(
+                true_classes=true_classes_mlp,
+                predicted_classes=predicted_classes_mlp,
+                ztf_test_names=ztf_test_names,
+            )
 
-        train_names = names[train_index]
-        val_names = names[val_index]
+        self.log_metrics_to_file(
+            num_epochs=num_epochs,
+            true_classes=true_classes_mlp,
+            prob_above_07=prob_above_07_mlp,
+            test_f1_score=test_f1_score,
+            test_acc=test_acc,
+            val_loss_avg=valid_loss_avg,
+        )
 
-        train_labels = labels[train_index]
-        val_labels = labels[val_index]
+    def generate_train_data(self, names, labels, redshifts, train_indices):
+        """Creates training and validation data, oversampling when needed.
+
+        Parameters
+        ----------
+        names The full list of ZTF objects.
```

Comment:
These param lists don't quite meet the numpy format. See rendered param blocks in https://superphot-plus--124.org.readthedocs.build/en/124/autoapi/src/superphot_plus/classify_ztf/index.html#src.superphot_plus.classify_ztf.CrossValidationTrainer.generate_test_data

Reply:
Thanks! I am used to writing docstrings using the numpy notatio, but for some reason I missed these.